tests/tcp_udp_listener/tcp_udp_listener.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#define CATCH_CONFIG_RUNNER
+
+#include "catch_wrapper.hpp"
+#include "socket_helper.h"
+
+std::string _protocol;
+uint16_t _local_port;
+
+volatile static LONG _global_counter = 0;
+
+static IPPROTO
+_get_protocol_from_string(std::string protocol)
+{
+    if (protocol.compare("tcp") == 0 || protocol.compare("TCP") == 0) {
+        return IPPROTO_TCP;
+    }
+
+    return IPPROTO_UDP;
+}
+
+void
+create_listener(_In_ receiver_socket_t* receiver_socket)
+{
+    _global_counter++;
+    // Post a receive. Wait for client to connect.
+    printf("=====================================\n");
+    printf("Posting async receive, counter = %d\n", _global_counter);
+    receiver_socket->post_async_receive();
+    printf("Waiting for receive to complete.\n");
+    receiver_socket->complete_async_receive(WSA_INFINITE, false);
+    printf("Received data from remote\n");
+
+    PSOCKADDR local_address = nullptr;
+    int local_length = 0;
+    receiver_socket->get_local_address(local_address, local_length);
+
+    // Send a response back.
+    std::string response = SERVER_MESSAGE + std::to_string(_local_port);
+    printf("Sending response: %s\n", response.c_str());
+    receiver_socket->send_async_response(response.c_str());
+    receiver_socket->complete_async_send(1000);
+    printf("Sent data to remote\n");
+}
+
+void
+create_tcp_listener(uint16_t local_port)
+{
+    stream_server_socket_t receiver_socket(SOCK_STREAM, IPPROTO_TCP, local_port);
+    while (true) {
```

Comment:
What causes this loop to exit?  Do the tests kill this process?  If so, maybe add a comment explaining.
Same question on line 62.

Reply:
Yes the test / user needs to kill the process. Updated comment mentioning the same.